Source/WebCore/platform/mock/GeolocationClientMock.h

```diff
@@ -44,13 +45,15 @@ class GeolocationController;
 
 // FIXME: this should not be in WebCore. It should be moved to WebKit.
 // Provides a mock object for the geolocation client.
-class GeolocationClientMock : public GeolocationClient {
+class GeolocationClientMock : public GeolocationClient, public RefCounted<GeolocationClientMock> {
     WTF_MAKE_FAST_ALLOCATED;
-    WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR(GeolocationClientMock);
 public:
     GeolocationClientMock();
     virtual ~GeolocationClientMock();
 
+    void ref() const final { RefCounted::ref(); }
```

Comment:
`GeolocationClientMock` needs to be private and you need to add a public `create()` function that returns a `Ref`.